lib/server.js

```diff
@@ -210,8 +212,42 @@ module.exports = {
       const parts = path.split(' ')
 
       let methods = []
-      if (route.config && route.config.pre && route.config.pre.length > 0) {
-        methods = methods.concat(route.config.pre)
+      if (route.config) {
+        // add validation
+        if (route.config.validate && Object.keys(route.config.validate).length > 0) {
+
+            route.config.validate.body = route.config.validate.payload // hapi compatibiluty
+            const validation = route.config.validate
+            const types = ['headers', 'params', 'query', 'body']
+            types.forEach((type) => {
+                let rule = validation[type]
+
+                // null, undefined, true - anything allowed
+                // false - nothing allowed
+                // {...} - ... allowed
+                 rule =  (rule === false ? Joi.object({}).allow(null)
+                : typeof rule === 'function' ? rule
+                : !rule || rule === true ? Joi.any()
+                : Joi.compile(rule));
+                validation[type] = rule;
+            })
+
+            methods = methods.concat((req,res, next) => {
+                // validate request
+                Joi.validate({headers: req.headers, params: req.params,query: req.query, body: req.body}, validation, (err, result)  =>{
```

Comment:
Hapi's doc say the validation is in this order `headers, params, query, payload` can you do the same ? Because if you validation all at the same time we don't know what will be validate first and joi return the first error.